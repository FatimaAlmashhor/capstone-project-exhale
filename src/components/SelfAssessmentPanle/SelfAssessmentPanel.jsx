/* eslint-disable jsx-a11y/control-has-associated-label */
import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

export default function SelfAssessmentPanel({ task, onClick }) {
  const { t } = useTranslation();
  const handleClick = (e) => {
    onClick(e);
  };
  const handleHover = (e) => {
    e.target.className += ' hover:bg-green-300';
  };

  const style = {
    buttonStyle:
      'bg-transparent border-2 border-gray-400 rounded-full focus:outline-none',
    trans: 'transition duration-500',
    hover: '',
  };

  return (
    <div className="transition-all duration-100 bg-background-primary">
      <div className="flex-auto h-64 py-10 lg:h-48">
        <h1
          className={` text-4xl text-textColor-primary text-center font-sans ${style.trans} ${style.hover} transition-all duration-100 `}
        >
          {t(task)}
        </h1>
      </div>
      <div className="flex-auto pt-20 ">
        <div className={` flex mt-8 mx-auto justify-center ${style.trans}`}>
          <div className="flex flex-col self-center justify-center flex-1 ">
            <button
              type="button"
              data-rate="20"
              className={` p-6  mt-6 mx-auto md:p-10 ${style.buttonStyle} ${style.trans} ${style.hover}`}
              onClick={handleClick}
              onMouseEnter={handleHover}
            />
            <p className="mx-auto mt-3 text-gray-500">{t('Never')}</p>
          </div>
          <div className="flex self-center justify-center flex-1 ">
            <button
              type="button"
              data-rate="40"
              className={` p-4  md:p-8 ${style.buttonStyle} ${style.trans} ${style.hover}`}
              onClick={handleClick}
              onMouseEnter={handleHover}
            />
          </div>
          <div className="flex self-center justify-center flex-1 ">
            <button
              type="button"
              data-rate="60"
              className={` p-3  md:p-6 ${style.buttonStyle} ${style.trans}${style.hover}`}
              onClick={handleClick}
              onMouseEnter={handleHover}
            />
          </div>
          <div className="flex self-center justify-center flex-1 ">
            <button
              type="button"
              data-rate="80"
              className={` p-4  md:p-8 ${style.buttonStyle} ${style.trans} ${style.hover}`}
              onClick={handleClick}
              onMouseEnter={handleHover}
            />
          </div>
          <div className="flex flex-col self-center justify-center flex-1 ">
            <button
              type="button"
              data-rate="100"
              className={`p-6  mt-6 mx-auto md:p-10 ${style.buttonStyle} ${style.trans} ${style.hover} `}
              onClick={handleClick}
              onMouseEnter={handleHover}
            />
            <p className="mx-auto mt-3 text-gray-500">{t('Always')}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
SelfAssessmentPanel.propTypes = {
  task: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
