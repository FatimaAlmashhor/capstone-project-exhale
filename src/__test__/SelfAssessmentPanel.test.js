import React from 'react';
import renderer from 'react-test-renderer';
import SelfAssessmentPanel from '../components/SelfAssessmentPanle';

it('renders correctly', () => {
  let fun = function(){ }
  const tree = renderer
    .create(<SelfAssessmentPanel task = {' '} onClick = {fun}/> )
    .toJSON();
  expect(tree).toMatchSnapshot();
});