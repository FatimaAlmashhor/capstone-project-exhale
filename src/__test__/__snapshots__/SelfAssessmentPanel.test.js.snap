// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="transition-all duration-100 bg-background-primary"
>
  <div
    className="flex-auto h-64 py-10 lg:h-48"
  >
    <h1
      className=" text-4xl text-textColor-primary text-center font-sans transition duration-500  transition-all duration-100 "
    >
       
    </h1>
  </div>
  <div
    className="flex-auto pt-20 "
  >
    <div
      className=" flex mt-8 mx-auto justify-center transition duration-500"
    >
      <div
        className="flex flex-col self-center justify-center flex-1 "
      >
        <button
          className=" p-6  mt-6 mx-auto md:p-10 bg-transparent border-2 border-gray-400 rounded-full focus:outline-none transition duration-500 "
          data-rate="20"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        />
        <p
          className="mx-auto mt-3 text-gray-500"
        >
          Never
        </p>
      </div>
      <div
        className="flex self-center justify-center flex-1 "
      >
        <button
          className=" p-4  md:p-8 bg-transparent border-2 border-gray-400 rounded-full focus:outline-none transition duration-500 "
          data-rate="40"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        />
      </div>
      <div
        className="flex self-center justify-center flex-1 "
      >
        <button
          className=" p-3  md:p-6 bg-transparent border-2 border-gray-400 rounded-full focus:outline-none transition duration-500"
          data-rate="60"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        />
      </div>
      <div
        className="flex self-center justify-center flex-1 "
      >
        <button
          className=" p-4  md:p-8 bg-transparent border-2 border-gray-400 rounded-full focus:outline-none transition duration-500 "
          data-rate="80"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        />
      </div>
      <div
        className="flex flex-col self-center justify-center flex-1 "
      >
        <button
          className="p-6  mt-6 mx-auto md:p-10 bg-transparent border-2 border-gray-400 rounded-full focus:outline-none transition duration-500  "
          data-rate="100"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        />
        <p
          className="mx-auto mt-3 text-gray-500"
        >
          Always
        </p>
      </div>
    </div>
  </div>
</div>
`;
